src/DateTimePicker.jsx

```diff
@@ -0,0 +1,240 @@
+		this.setState({ isDateTimeLocalSupported: this.hasBrowserSupport() });
+	}
+
+	/**
+	* @function hasBrowserSupport
+	* @description test if this browser supports datetime local
+	* @return bool whether or not this browser supports datetime local
+	*/
+	hasBrowserSupport() {
+		const input = document.createElement('input'),
+			invalidValue = 'notadate';
+
+		// some browsers (as Android stock browsers) pretend they support
+		// certain input types,
+		// so we set the value to an invalid value and see if browser rejects
+
+		input.setAttribute('type', 'datetime-local');
+		input.setAttribute('value', invalidValue);
+
+		return !(this.props.forceCalendar || input.value === invalidValue);
+	}
+
+	/**
+	* @function getDate
+	* @description gets a date from the datetime value in state
+	* @return Date
+	*/
+	getDate() {
+		if (!this.state.datetime) {
+			return;
+		}
+		const datetime = new Date(this.state.datetime);
+		return new Date(datetime.getFullYear(), datetime.getMonth(), datetime.getDate());
+	}
+
+	/**
+	* @function setDate
+	* @description sets the date portion of the state's datetime
+	* @param Date value date to set in state
+	*/
+	setDate(value) {
+		const newDate = new Date(value);
+		const datetime = new Date(this.state.datetime);
+
+		datetime.setFullYear(newDate.getFullYear(), newDate.getFullMonth(), newDate.getDate());
+		this.setState({ datetime });
+	}
+
+	/**
+	* @function parseTimeFromDateTime
+	* @description gets the time string part of our datetime state
+	* @return String
+	*/
+	parseTimeFromDateTime(datetime) {
+		return (datetime.toTimeString().split(' ')[0]).split(':', 2).join(':'); // TODO localize toLocaleTimeString ?
+	}
+
+	/**
+	* @function getTime
+	* @description gets the time portion of the state's datetime
+	* @return String time portion of datetime  (hour and min)
+	*/
+	getTime() {
+		if (!this.state.datetime) {
+			return;
+		}
+		const datetime = new Date(this.state.datetime);
+		// split on space and leave off timezone and seconds
+		return this.parseTimeFromDateTime(datetime);
+	}
+
+	/**
+	* @function setTime
+	* @description sets the time portion of the state's datetime
+	* @param value String valid time string
+	*/
+	setTime(value) {
+		const newTime = new Date(value);
+		const datetime = new Date(this.state.datetime);
+
+		datetime.setTime(newTime.getTime());
+		this.setState({ datetime });
+	}
+
+	setDateTime(value) {
+		this.setState({ datetime: new Date(value) });
+	}
+
+	/**
+	* @function onFocus
+	* @param Event e
+	* @description called if datetime compound component and used to set focused
+	* class on combo component
+	*/
+	onFocus(e) {
+		this.backgroundEl.classList.add('focused');
+	}
+
+	/**
+	* @function onBlur
+	* @param Event e
+	* @description called when either of the date or time inputs is blurred
+	* takes off focus class when neither of them are in focus
+	*/
+	onBlur(e) {
+		const timeInput = ReactDOM.findDOMNode(this.timeComponent).querySelector('input[type=time]');
```

Comment:
yeah, there are a couple problems with the current implementation. First, you're manipulating the DOM through `this.backgroundEl.classList` in `onFocus` and `onBlur` rather than using `this.setState` to trigger a re-render and set the right `className` in JSX. You shouldn't ever directly manipulate the DOM in React - use the component's render method and read from `this.state` and `this.props`. If you don't do that, React could re-render your component and blow away any changes that you've manually applied.

To a slightly lesser extent, the `querySelector` usage is also problematic because you are accessing the DOM independently of React - you can actually use the `ref` that you've already set up for `CalendarComponent` to access its `<input>` (`this.dateComponent.node` will get you there), and you can add a `ref` to the time `<input>` in `TimeInput` so that you can access it with something like `this.timeComponent.node` to avoid having to use `querySelector`.